KsApi/models/ValidateCheckoutEnvelope.swift

```diff
@@ -14,6 +13,11 @@ extension ValidateCheckoutEnvelope {
     guard let envelope = ValidateCheckoutEnvelope.validateCheckoutEnvelope(from: data) else {
       return SignalProducer(error: .couldNotParseJSON)
     }
+
+    guard data.checkout?.isValidForOnSessionCheckout.valid == true else {
+      return SignalProducer(error: .validateCheckoutError(envelope.messages.first ?? ""))
```

Comment:
Nit: I think the message param here should be optional instead of passing in the empty string, so that we can show the "Something went wrong" default (instead of a banner with an empty string) if the server doesn't give us an error message for some weird reason. Or we could just pass in the entire list of messages instead; that might make more sense actually.